### DISCUSSION TITLE
How to properly configure the title on home page?

### DISCUSSION THREAD

**Discussion by jcubic** [facebook/docusaurus#9996] [Q&A] [answered]
I have a website, it shows title as `LIPS Scheme | LIPS Scheme` on home page.

I only set it once:

```javascript
const config: Config = {
  title: 'LIPS Scheme',
  tagline: 'Powerfull Scheme based Lisp in JavaScript',
  favicon: 'img/favicon.ico',

```

**Comment by slorber** [comment]
Show me a repro: https://docusaurus.new/stackblitz

**Reply by jcubic** [reply]
npm run build doesn't work:

```
npm run build

> docusaurus@0.0.0 build
> docusaurus build

jsh: command not found: docusaurus
```

**Reply by jcubic** [reply]
Start also doesn't work:
```
yarn install v1.22.19
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
[3/5] 🚚  Fetching packages...
error An unexpected error occurred: "Failed to fetch dependencies:
- @types/gtag.js@0.0.12".
info If you think this is a bug, please open a bug report with the information provided in "/home/projects/jvmiwbnqk.github/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Reply by OzakIOne** [reply]
This may be related to an extension blocking the gtag module, try to disable any privacy related extension or in an other browser with bare minimum extensions

**Reply by jcubic** [reply]
Sorry, but I have no idea how to disable privacy protection in Brave that is privacy focused. Will try different browser when will have a next problem. Thanks.

**Reply by OzakIOne** [reply]
Disabling [brave shield](https://brave.com/shields/) should do the work (and other adblocker if you also have one in parallel)

**Comment by jcubic** [comment, marked as answer]
So the problem is with the starter code:
```
export default function Home() {
  const {siteConfig} = useDocusaurusContext();
  return (
    <Layout
      title={`Hello from ${siteConfig.title}`}
      description="Description will go into a meta tag in <head />">
      <HomepageHeader />
      <main>
        <HomepageFeatures />
      </main>
    </Layout>
  );
}
```

This create title `"Hello from ${siteConfig.title} | ${siteConfig.title}"` According to  #7535 it's not possible to remove `| ${siteConfig.title}` so you should not use `siteConfig.title` it the Layout title at all.

**Reply by slorber** [reply]
Having `Hello from my Site | My Site` title is done on purpose in the starter template so that you can figure it out intuitively how title templating works. You can omit the title prop if you want to and the homepage title will just be your site title.

I knew what the problem was, and only asked for a repro because when you provide a repro, you usually figure out the problem on your own. For that reason, always provide a repro before opening an issue or a discussion.

If StackBlitz does not work well in brave you can just create a GitHub repository or use CodeSandbox instead.

**Comment by maheshj01** [comment]
@slorber is it possible to update the title on the navbar and link in runtime?

I have a nested  documentation sites and would like to change the title on the nested pages

**Reply by maheshj01** [reply]
see https://docs.maheshjamdade.com/ when we navigate to internal pages (navabr_router or git guide) I would like to change the title on navbar and how to show the banner only on internal pages not home page?

**Reply by slorber** [reply]
It's difficult to understand. Please be very concrete. Which page should have which title exactly.

**Reply by maheshj01** [reply]
Apologies @slorber, Let me rephrase that

I have a docs website https://docs.maheshjamdade.com/ this page has a Navbar titled "Mahesh's Docs which is fine", But when I navigate inside by clicking the card on those pages to https://docs.maheshjamdade.com/git-tutorial

I would like to change the Navbar title from "Mahesh;s Docs" to something like "Git guide"

So my question is, is it possible to change the default Navbar;s title programmatically?

And secondly, This page https://docs.maheshjamdade.com/ has a yellow banner at the top is it possible to show it on those internal pages https://docs.maheshjamdade.com/git-tutorial

**Reply by jcubic** [reply]
@maheshmnj You can [swizzle](https://docusaurus.io/docs/swizzling/) the NavBar and customize it.

**Reply by slorber** [reply]
The page https://docs.maheshjamdade.com/git-tutorial/ has title "Overview" like the title of the doc it renders.

If you want it to be something else, you can swizzle the DocItemMetadata component and pass it the title you want there:

```tsx
export default function DocItemMetadata(): JSX.Element {
  const {metadata, frontMatter, assets} = useDoc();
  return (
    <PageMetadata
      title={metadata.title}
      description={metadata.description}
      keywords={frontMatter.keywords}
      image={assets.image ?? frontMatter.image}
    />
  );
}
```

For the navbar, you can swizzle the theme layout or navbar.
